packages/opentelemetry-plugin-http/src/http.ts

```diff
@@ -0,0 +1,437 @@
+      }
+
+      const rootSpan = plugin._tracer.startSpan(path, spanOptions);
+      return plugin._tracer.withSpan(rootSpan, () => {
+        plugin._tracer.scopeManager.bind(request);
+        plugin._tracer.scopeManager.bind(response);
+
+        // Wraps end (inspired by:
+        // https://github.com/GoogleCloudPlatform/cloud-trace-nodejs/blob/master/src/plugins/plugin-connect.ts#L75)
+        const originalEnd = response.end;
+        response.end = function(this: ServerResponse, ...args: ResponseEndArgs) {
+          response.end = originalEnd;
+          // Cannot pass args of type ResponseEndArgs, Expected 1-2 arguments, but got 1 or more.
+          // tslint:disable-next-line:no-any
+          const returned = response.end.apply(this, arguments as any);
+          const requestUrl = request.url ? url.parse(request.url) : null;
+          const host = headers.host || 'localhost';
+          const userAgent = (headers['user-agent'] || headers['User-Agent']) as string;
+
+          rootSpan.setAttribute(HttpPlugin.ATTRIBUTE_HTTP_HOST, host.replace(/^(.*)(\:[0-9]{1,5})/, '$1'));
+
+          rootSpan.setAttribute(HttpPlugin.ATTRIBUTE_HTTP_METHOD, method);
+          if (requestUrl) {
+            rootSpan.setAttribute(HttpPlugin.ATTRIBUTE_HTTP_PATH, requestUrl.pathname || '');
+            rootSpan.setAttribute(HttpPlugin.ATTRIBUTE_HTTP_ROUTE, requestUrl.path || '');
+          }
+          if (userAgent) {
+            rootSpan.setAttribute(HttpPlugin.ATTRIBUTE_HTTP_USER_AGENT, userAgent);
+          }
+          rootSpan.setAttribute(HttpPlugin.ATTRIBUTE_HTTP_STATUS_CODE, response.statusCode.toString());
+
+          rootSpan.setStatus(HttpPlugin.parseResponseStatus(response.statusCode));
+
+          if (plugin.options.applyCustomAttributesOnSpan) {
+            plugin.options.applyCustomAttributesOnSpan(rootSpan, request, response);
+          }
+
+          rootSpan.end();
+          return returned;
+        };
+        return original.apply(this, [event, ...args]);
+      });
+    };
+  }
+
+  private outgoingRequestFunction(original: Func<ClientRequest>, plugin: HttpPlugin): Func<ClientRequest> {
+    return function outgoingRequest(
+      this: {},
+      options: RequestOptions | string,
+      callback?: HttpCallback
+    ): ClientRequest {
+      if (!options) {
+        return original.apply(this, [options, callback]);
+      }
+
+      // Makes sure the url is an url object
+      let pathname;
+      let method;
+      let origin = '';
+      if (typeof options === 'string') {
+        const parsedUrl = url.parse(options);
+        options = parsedUrl;
+        pathname = parsedUrl.pathname || '';
+        origin = `${parsedUrl.protocol || 'http:'}//${parsedUrl.host}`;
+      } else {
+        try {
+          pathname = (options as url.URL).pathname;
+          if (!pathname) {
+            pathname = options.path ? url.parse(options.path).pathname : '';
+          }
+          method = options.method;
```

Comment:
This can be `method  = options.method ? options.method.toUpperCase() : 'GET';` and remove https://github.com/open-telemetry/opentelemetry-js/pull/161/files#diff-d1b28358da758893ac27e367738e7e51R403 block.